app/services/patient_import/spreadsheet_transformer.rb

```diff
@@ -0,0 +1,304 @@
+        row[:last_visit_medication_5]
+      ]
+
+      drug_names.map { |name|
+        medication(
+          name: name,
+          patient_id: patient_id,
+          created_at: row[:last_visit_date]
+        )
+      }.compact
+    end
+
+    def patient_id(row)
+      PatientBusinessIdentifier.find_by(
+        identifier_type: row[:identifier_type],
+        identifier: row[:identifier]
+      )&.patient_id || SecureRandom.uuid
+    end
+
+    def business_identifier_metadata_version(row)
+      case row[:identifier_type]
+      when "simple_bp_passport"
+        "org.simple.bppassport.meta.v1"
+      else
+        "org.simple.#{row[:identifier_type]}.meta.v1"
+      end
+    end
+
+    def registration_facility_id
+      facility.id
+    end
+
+    def import_user
+      @import_user = PatientImport::ImportUser.find_or_create
+    end
+
+    def create_import_user
+      user = User.new(
+        full_name: "import-user",
+        organization_id: Organization.take,
+        device_created_at: Time.current,
+        device_updated_at: Time.current
+      )
+      phone_number_authentication = PhoneNumberAuthentication.new(
+        phone_number: IMPORT_USER_PHONE_NUMBER,
+        password: "#{rand(10)}#{rand(10)}#{rand(10)}#{rand(10)}",
+        registration_facility_id: facility.id
+      ).tap do |pna|
+        pna.set_otp
+        pna.invalidate_otp
+        pna.set_access_token
+      end
+
+      user.phone_number_authentications = [phone_number_authentication]
+      user.sync_approval_denied("bot user for import")
+      user.save!
+
+      user
+    end
+
+    def patient_status(row)
+      row[:died] == "yes" ? :dead : :active
+    end
+
+
+
+    def medication(name:, patient_id:, created_at:)
+      medication_record = medication_by_name_dosage_and_frequency(name) || medication_by_name_and_dosage(name)
+
+      return nil unless medication_record.present?
+
+      {
+        id: SecureRandom.uuid,
+        patient_id: patient_id,
+        facility_id: registration_facility_id,
+        name: medication_record.name,
+        rxnorm_code: medication_record.rxnorm_code,
+        dosage: medication_record.dosage,
+        is_protocol_drug: protocol_drug?(medication_record),
+        created_at: timestamp(created_at),
+        updated_at: timestamp(created_at),
+        is_deleted: false
+      }
+    end
+
+    def protocol_drug?(medication)
+      facility.protocol.protocol_drugs.exists?(name: medication.name, dosage: medication.dosage)
+    end
+
+    def localized_frequency(medication)
+      return nil unless medication.frequency.present?
+      I18n.translate("helpers.label.drug.frequency.#{medication.frequency}", locale: CountryConfig.current[:dashboard_locale])
+    end
+
+    def medication_by_name_and_dosage(name)
+      return unless name.present?
+
+      Medication.all.to_a.find do |medication|
+        name.gsub(/\s+/, "") == "#{medication.name}#{medication.dosage}".gsub(/\s+/, "")
```

Comment:
Who is going to make sure medications in the spreadsheet actually match the values we have in our dataset?  This seems like it could be pretty difficult to keep in sync.

Reply:
It's not great, but we (Meg primarily) manages the spreadsheet that we share to facilities. The medication fields have strict Excel data validations on them. If we update our `medications.yml` list, we need to notify Meg.

Is there an automatic Github thing we can set up to remind us to do this?

Reply:
If Meg gets a GitHub account, we could add that specific file to the CODEOWNERS file I've started over in https://github.com/simpledotorg/simple-server/pull/2467.  

Or we could have someone else as the 'owner' of that file who would ensure Meg gets a heads up when it is changed.

Reply:
I'll add us to the codeowners, and a comment to the YML file with instructions on who to notify.

Reply:
I've added a comment to the `medications.yml` file with instructions, and I'll add the file itself to the codeowners list in #2467 directly.